packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -419,11 +419,42 @@ export class ProjectDataSource {
 
   async defaultSpecFileName (): Promise<string> {
     const { specPattern = [] } = await this.ctx.project.specPatterns()
+    let fileExtensionToUse: FileExtension = this.ctx.lifecycleManager.fileExtensionToUse
+
+    // If generating a component test then check whether there are JSX/TSX files present in the project.
+    // If project uses JSX then user likely wants to use JSX for their tests as well.
+    // JSX can be used (or not used) with a variety of frameworks depending on user preference/config, so
+    // the only reliable way to determine is whether there are files with JSX extension present
```

Comment:
Ultimately it would be nice to put the extension preference for generated tests in the component testing config or something so there is the ability to override this if needed 🤔. But I think this is a good rubric, especially for new people to get what they probably expect.